### PR TITLE
MINOR: Fix Findbugs issue about useless object in KafkaAdminClient

### PR DESCRIPTION
**Pull request by attila-s**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

Fix Findbugs issue

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

org.apache.kafka.clients.admin.KafkaAdminClientTest unit tests passed.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -1517,11 +1517,9 @@ void handleFailure(Throwable throwable) {
     @Override
     public DescribeConfigsResult describeConfigs(Collection<ConfigResource> configResources, final DescribeConfigsOptions options) {
         final Map<ConfigResource, KafkaFutureImpl<Config>> unifiedRequestFutures = new HashMap<>();
-        final Map<ConfigResource, KafkaFutureImpl<Config>> brokerFutures = new HashMap<>(configResources.size());
-
         // The BROKER resources which we want to describe.  We must make a separate DescribeConfigs
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:1520]: I guess we should update or remove this comment?

**Review by hachikuji — COMMENTED** [review]
Thanks for the patch. Left a minor comment.

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java (hunk 1 above)
Comment (asasvari) [reply clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:1520]: @hachikuji thanks for the review. I believe that comment about ``unifiedRequestResources`` is still valid, see:

```java
        // The non-BROKER resources which we want to describe.  These resources can be described by a
        // single, unified DescribeConfigs request.
        final Collection<Resource> unifiedRequestResources = new ArrayList<>(configResources.size());

        for (ConfigResource resource : configResources) {
            if (resource.type() == ConfigResource.Type.BROKER && !resource.isDefault()) {
                brokerFutures.put(resource, new KafkaFutureImpl<Config>());
            } else {
                unifiedRequestFutures.put(resource, new KafkaFutureImpl<Config>());
                unifiedRequestResources.add(configResourceToResource(resource));
            }
        }
```

**Comment by chia7712** [comment]
@attila-s Thanks for your patch. This issue was fixed by https://github.com/apache/kafka/commit/49db5a63c043b50c10c2dfd0648f8d74ee917b6a. please feel free to ping me if you have other pending PRs.

**Pull request closed without merging**